packages/flutter_tools/lib/src/base/deferred_component.dart

```diff
@@ -0,0 +1,159 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io' as io;
+
+import 'package:meta/meta.dart';
+
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../globals.dart' as globals;
+import '../project.dart';
+import '../build_info.dart';
+import '../build_system/build_system.dart';
+
+// TODO: Document classes and methods in this file. Fix privacy.
+// Add floating methods to classes to de-pollute namespace
+
+/// Represents a configured deferred component as defined in
+/// the app's pubspec.yaml.
+class DeferredComponent {
+  /// The name of the deferred component. There should be a matching
+  /// android dynamic feature module with the same name.
+  String name;
+
+  /// The dart libraries this component includes. These libraries are
+  /// cross referenced against gen_snapshot's loading unit manifest to
+  /// determine which loading units are included in this component.
+  List<String> libraries;
```

Comment:
I'm still not sure I understood what the answer to my question was. If the loading units have a list of their libraries, why do you need to keep another list of the libraries?